Assets/Scripts/AttackLogic.cs

```diff
@@ -0,0 +1,196 @@
+
+    private float _timer;
+    private GameObject _target;
+    private float _distanceToTarget;
+
+    // ----temporary as animation replacement------
+        private MeshRenderer _gameObjectRenderer;
+        private Material _prevMat;
+        private Material _attackAnimationMaterial;
+    // ---------
+
+    private void Awake()
+    {
+        // init components
+        _movable = GetComponent<IMovable>();
+
+        // -----temp replacement for animation-----
+            _gameObjectRenderer = gameObject.GetComponentInChildren<MeshRenderer>();
+            _prevMat = _gameObjectRenderer.material;
+            _attackAnimationMaterial = new Material(Shader.Find("Standard"));
+            _attackAnimationMaterial.color = Color.yellow;
+        // ----------
+    }
+
+    private void OnEnable()
+    {
+        Status = AttackStatus.None;
+    }
+
+    /// <summary>
+    /// If it has a target, attacks it.
+    /// </summary>
+    private void Update()
+    {
+        if (_target != null)
+        {
+            Attack();
+        }
+    }
+
+    /// <summary>
+    /// Attacks the target. If the target is not in attacking range, chases it. If the target is near
+    /// performs a hit on it. If currently performing a hit, waits for finishing the hit and deals 
+    /// damage if hit was successful. Either resets afterwards or continues chasing / attacking target.
+    /// </summary>
+    private void Attack()
+    {
+        _distanceToTarget = Vector3.Distance(_target.transform.position, transform.position);
+        if (_distanceToTarget < attackRange && Status == AttackStatus.None)
+        {
+            IsInRange();
+        }
+        else if (Status == AttackStatus.NotFinished)
+        {
+            Status = PerformHit();
+            if (Status != AttackStatus.NotFinished)
+            {
+                HitFinished();
+            }
+        }
+        else
+        {
+            // chase target
+            _movable.Move(_target.transform.position);
+        }
+    }
+
+    /// <summary>
+    /// Stops moving, faces the target if not facing yet, if facing, starts a hit.
+    /// </summary>
+    private void IsInRange()
+    {
+        _movable.StopMoving();
+        // face target
+        if (_movable.FaceTarget(_target, true))
+        {
+            // faces target
+            // perform hit
+            Status = AttackStatus.NotFinished;
+
+            // -----temp-----
+            _prevMat = _gameObjectRenderer.material;
+            _gameObjectRenderer.material = _attackAnimationMaterial;
+            // ----------
+        }
+    }
+
+    /// <summary>
+    /// Ends the hit. Deals damage to the target, if hit was successful. Resets aggro if necessary.
+    /// </summary>
+    private void HitFinished()
+    {
+        // hit animation performed
+        if (Status == AttackStatus.Hit)
+        {
+            // deal damage
+            IDamageable damageable = (IDamageable)_target.GetComponent<IDamageable>();
```

Comment:
I'm pretty sure this cast ist redundant, since GetComponent<IDamageable> returns an object of type IDamageable 🤔

Reply:
for sure, ye I forgot that here